### ISSUE TITLE
Active response don't work for firewall-drop

### ISSUE THREAD

**Issue by Linow974** [wazuh/wazuh#11842]
Hello,

I tried setting up wazuh's "Firewall-drop", but it doesn't work.

Here is my ossec.conf file:

`  <command>
    <name>firewall-drop</name>
    <executable>firewall-drop</executable>
    <expect>srcip</expect>
    <timeout_allowed>yes</timeout_allowed>
  </command>

....

  <active-response>
    <command>firewall-drop</command>
    <location>all</location>
    <rules_id>5710,5712,5716,100100</rules_id>
    <timeout>1800</timeout>
  </active-response>

`

I noticed with the command : `/var/ossec/active-response/bin/firewall-drop add - 192.168.x.x` that the script provided was not good, because the command was going around in circles, doing nothing.
So I installed the firewall-drop.sh script, old version and I ran : `/var/ossec/active-response/bin/firewall-drop.sh add - 192.168.x.x`
And when I do iptables -L, I get my blocked IP fine.
However, after replacing this:

`<executable>firewall-drop.sh</executable>`

that doesn't work either.

I did a lot of research on the problem, I tried to "debug" via the "internal_options.conf" file. I could see that active response was not even executed, whatever the type of rule sought (after having attempted an ssh attack: ```for i in `seq 1 10`; do sshpass -p 'wrong_password' ssh -o StrictHostKeyChecking=no 192.168.x.x; done``` )

Then I tried to do the same having "pf-block". And this worked : 

`<command>
  <name>pf-block</name>
  <executable>pf</executable>
</command>

....

<active-response>
  <command>pf-block</command>
  <location>defined-agent</location>
  <agent_id>001</agent_id>
  <rules_group>authentication_failed|authentication_failures</rules_group>
</active-response>`

It worked with an error, but active response was enabled! Here are the logs: active-response.log

`2022/01/17 15:28:50 active-response/bin/pf: Cannot read 'srcip' from data
2022/01/17 15:51:22 active-response/bin/pf: Starting`

I tried firewall with the same syntax, still nothing.

I'm a little lost... Could someone help me?

--------

Wazuh version : 4.2.0-5.1

**Comment by palaciosjeremias** [comment]
Hello Linow974,
Can you share with us the complete configuration file of your agent?
If you are using Wazuh 4.2, please keep in mind that some configurations have been deprecated. [link ](https://documentation.wazuh.com/current/user-manual/reference/ossec-conf/commands.html)
The command that you are testing should work correctly with the below configuration:
```
<command>
  <name>firewall-drop</name>
  <executable>firewall-drop</executable>
  <timeout_allowed>yes</timeout_allowed>
</command>
....
 <active-response>
  <command>firewall-drop</command>
  <location>all</location>
  <rules_id>5710,5712,5716,100100</rules_id>
  <timeout>1800</timeout>
 </active-response>
```

I will wait for your response.
Please let me know if this helps. If you have further doubts don´t hesitate to ask.
Best regards.

**Comment by Linow974** [comment]
Hello, thank you for your help!

I replaced the good configurations, indeed <expect> is obsolete, but you should know that I tried all types of configurations in order to try to find the problem.

So, I perform the following attack on the host containing a wazuh agent :

```
ssh us...@192.168.x.x
us...@192.168.x.x's password:
```

In debug mode, this is what I get for `ossec.log` :

```
2022/01/19 08:18:58 wazuh-remoted[3014] manager.c:1058 at lookfor_agent_group(): DEBUG: Agent '010' group is 'default'
2022/01/19 08:18:58 wazuh-remoted[3014] manager.c:1144 at read_controlmsg(): DEBUG: read_controlmsg(): reading 'Linux |cible |5.4.128-1-pve |#1 SMP PVE 5.4.128-2 (Wed, 18 Aug 2021 16:20:02 +0200) |x86_64 [Debian GNU/Linux|debian: 10 (buster)] - Wazuh v4.2.5 / ab73af41699f13fdd81903b5f23d8d00
1715fb875296f5082db84216d4a850f8 merged.mg
#"_agent_ip":192.168.x.x
'
2022/01/19 08:18:59 wazuh-remoted[3014] state.c:59 at rem_write_state(): DEBUG: Updating state file.
2022/01/19 08:19:04 wazuh-remoted[3014] secure.c:314 at rem_keyupdate_main(): DEBUG: Checking for keys file changes.
2022/01/19 08:19:04 wazuh-remoted[3014] state.c:59 at rem_write_state(): DEBUG: Updating state file.
2022/01/19 08:19:04 wazuh-remoted[3014] manager.c:677 at c_files(): DEBUG: Updating shared files sums.
2022/01/19 08:19:04 wazuh-remoted[3014] manager.c:912 at c_files(): DEBUG: End updating shared files sums.
2022/01/19 08:19:08 wazuh-remoted[3014] secure.c:521 at HandleSecureMessage(): DEBUG: TCP socket 10 already in keystore. Updating...
2022/01/19 08:19:08 wazuh-remoted[3014] manager.c:242 at save_controlmsg(): DEBUG: save_controlmsg(): inserting 'Linux |cible |5.4.128-1-pve |#1 SMP PVE 5.4.128-2 (Wed, 18 Aug 2021 16:20:02 +0200) |x86_64 [Debian GNU/Linux|debian: 10 (buster)] - Wazuh v4.2.5 / ab73af41699f13fdd81903b5f23d8d00
1715fb875296f5082db84216d4a850f8 merged.mg
#"_agent_ip":192.168.x.x
```

And for the `alerts.log` :

```
** Alert 1642580325.8320: mail  - local,syslog,sshd,invalid_login,authentication_failed,pci_dss_10.2.4,pci_dss_10.2.5,pci_dss_10.6.1,gpg13_7.1,gdpr_IV_35.7.d,gdpr_IV_32.2,hipaa_164.312.b,nist_800_53_AU.14,nist_800_53_AC.7,nist_800_53_AU.6,tsc_CC6.1,tsc_CC6.8,tsc_CC7.2,tsc_CC7.3,
2022 Jan 19 08:18:45 (cible) 192.168.1.20->/var/log/auth.log

Rule: 5710 (level 13) -> 'sshd: Attempt to login using a non-existent user'
Src IP: 192.168.x.x
Src Port: 42444
Jan 19 08:18:45 cible sshd[1001]: Invalid user user from 192.168.x.x port 42444
```

I still have nothing in auth.log during an ssh attack, i have activate `logcollector.debug=2` in my `local_internal_options.conf`

`ossec.log` :

```
2022/01/19 08:23:55 wazuh-remoted[4196] state.c:59 at rem_write_state(): DEBUG: Updating state file.
2022/01/19 08:24:00 wazuh-remoted[4196] secure.c:314 at rem_keyupdate_main(): DEBUG: Checking for keys file changes.
2022/01/19 08:24:00 wazuh-remoted[4196] state.c:59 at rem_write_state(): DEBUG: Updating state file.
2022/01/19 08:24:00 wazuh-remoted[4196] manager.c:677 at c_files(): DEBUG: Updating shared files sums.
2022/01/19 08:24:00 wazuh-remoted[4196] manager.c:912 at c_files(): DEBUG: End updating shared files sums.
2022/01/19 08:24:03 wazuh-logcollector[4230] read_syslog.c:104 at read_syslog(): DEBUG: Reading syslog message: 'Jan 19 08:24:02 wazuh-manager filebeat[110]: 2022-01-19T08:24:02'...
2022/01/19 08:24:03 wazuh-logcollector[4230] read_syslog.c:148 at read_syslog(): DEBUG: Read 1 lines from /var/log/syslog
2022/01/19 08:24:04 wazuh-remoted[4196] secure.c:521 at HandleSecureMessage(): DEBUG: TCP socket 11 already in keystore. Updating...
2022/01/19 08:24:04 wazuh-remoted[4196] manager.c:242 at save_controlmsg(): DEBUG: save_controlmsg(): inserting 'Linux |cible |5.4.128-1-pve |#1 SMP PVE 5.4.128-2 (Wed, 18 Aug 2021 16:20:02 +0200) |x86_64 [Debian GNU/Linux|debian: 10 (buster)] - Wazuh v4.2.5 / ab73af41699f13fdd81903b5f23d8d00
1715fb875296f5082db84216d4a850f8 merged.mg
#"_agent_ip":192.168.1.20
```

I don't understand why the alerts.log file shows the path /var/log/auth.log when there is not the alert in this file, but in another...

While waiting for your answer, I studied an answer made here (the one at 13:39:42) https://groups.google.com/g/wazuh/c/pgfBgn4WaNQ/m/uWGDPfSFAAAJ

So I created a new one:

```
<rule id="100010" level="13">
  <match>test</match>
  <description>Testing active_response</description>
</rule>
</group>

```
With this conf :

```
  <active-response>
    <command>firewall-drop</command>
    <location>Local</location>
    <level>13</level>                        
    <timeout>1800</timeout>
  </active-response>
```

And if I run this command :

`echo "Nov  8 11:16:26 centos-manager1 sshd[20046]: test from 172.17.1.1 port 39588" >> /tmp/testing.log`

I have in `active-response.log` : 

```
2022/01/19 08:42:56 active-response/bin/firewall-drop: Starting
2022/01/19 08:42:56 active-response/bin/firewall-drop: {"version":1,"origin":{"name":"node01","module":"wazuh-execd"},"command":"add","parameters":{"extra_args":[],"alert":{"timestamp":"2022-01-19T08:42:56.529+0000","rule":{"level":13,"description":"sshd: Attempt to login using a non-existent user","id":"5710","mitre":{"id":["T1110"],"tactic":["Credential Access"],"technique":["Brute Force"]},"firedtimes":1,"mail":true,"groups":["local","syslog","sshd","invalid_login","authentication_failed"],"pci_dss":["10.2.4","10.2.5","10.6.1"],"gpg13":["7.1"],"gdpr":["IV_35.7.d","IV_32.2"],"hipaa":["164.312.b"],"nist_800_53":["AU.14","AC.7","AU.6"],"tsc":["CC6.1","CC6.8","CC7.2","CC7.3"]},"agent":{"id":"000","name":"wazuh-manager"},"manager":{"name":"wazuh-manager"},"id":"1642581776.30257","full_log":"Nov  8 11:16:26 centos-manager1 sshd[20046]: invalid user from 172.17.1.1 port 39588","predecoder":{"program_name":"sshd","timestamp":"Nov  8 11:16:26","hostname":"centos-manager1"},"decoder":{"parent":"sshd","name":"sshd"},"data":{"srcport":"39588"},"location":"/tmp/testing.log"},"program":"active-response/bin/firewall-drop"}}

2022/01/19 08:42:56 active-response/bin/firewall-drop: Cannot read 'srcip' from data
```

and in `alerts.log` : 
```

** Alert 1642581954.33783: mail  - local,syslog,sshd,
2022 Jan 19 08:45:54 centos-manager1->/tmp/testing.log
Rule: 100010 (level 13) -> 'Testing active_response'
Nov  8 11:16:26 centos-manager1 sshd[20046]: test from 172.17.1.1 port 39588

** Alert 1642581954.34023: - ossec,active_response,pci_dss_11.4,gpg13_4.13,gdpr_IV_35.7.d,nist_800_53_SI.4,tsc_CC6.1,tsc_CC6.8,tsc_CC7.2,tsc_CC7.3,tsc_CC7.4,
2022 Jan 19 08:45:54 wazuh-manager->/var/ossec/logs/active-responses.log
Rule: 651 (level 3) -> 'Host Blocked by firewall-drop Active Response'
2022/01/19 08:45:54 active-response/bin/firewall-drop: {"version":1,"origin":{"name":"node01","module":"wazuh-execd"},"command":"add","parameters":{"extra_args":[],"alert":{"timestamp":"2022-01-19T08:45:54.653+0000","rule":{"level":13,"description":"Testing active_response","id":"100010","firedtimes":1,"mail":true,"groups":["local","syslog","sshd"]},"agent":{"id":"000","name":"wazuh-manager"},"manager":{"name":"wazuh-manager"},"id":"1642581954.33783","full_log":"Nov  8 11:16:26 centos-manager1 sshd[20046]: test from 172.17.1.1 port 39588","predecoder":{"program_name":"sshd","timestamp":"Nov  8 11:16:26","hostname":"centos-manager1"},"decoder":{"name":"sshd"},"location":"/tmp/testing.log"},"program":"active-response/bin/firewall-drop"}}
version: 1
origin.name: node01
origin.module: wazuh-execd
command: add
parameters.extra_args: []
parameters.alert.timestamp: 2022-01-19T08:45:54.653+0000
parameters.alert.rule.level: 13
parameters.alert.rule.description: Testing active_response
parameters.alert.rule.id: 100010
parameters.alert.rule.firedtimes: 1
parameters.alert.rule.mail: true
parameters.alert.rule.groups: ["local", "syslog", "sshd"]
parameters.alert.agent.id: 000
parameters.alert.agent.name: wazuh-manager
parameters.alert.manager.name: wazuh-manager
parameters.alert.id: 1642581954.33783
parameters.alert.full_log: Nov  8 11:16:26 centos-manager1 sshd[20046]: test from 172.17.1.1 port 39588
parameters.alert.predecoder.program_name: sshd
parameters.alert.predecoder.timestamp: Nov  8 11:16:26
parameters.alert.predecoder.hostname: centos-manager1
parameters.alert.decoder.name: sshd
parameters.alert.location: /tmp/testing.log
parameters.program: active-response/bin/firewall-drop
```

So it works, but now you have to create an alert with an SRCIP field !

And if I try with an alert activating level 13 also:

`echo "Nov  8 11:16:26 centos-manager1 sshd[20046]: invalid user from 172.17.1.1 port 39588" >> /tmp/testing.log`

it also works !

And for auth.log : 

`echo "Nov  8 11:16:26 centos-manager1 sshd[20046]: invalid user from 172.17.1.1 port 39588" >> /var/log/auth.log`

Is the same : 

```
2022/01/19 08:50:26 active-response/bin/firewall-drop: Starting
2022/01/19 08:50:26 active-response/bin/firewall-drop: {"version":1,"origin":{"name":"node01","module":"wazuh-execd"},"command":"add","parameters":{"extra_args":[],"alert":{"timestamp":"2022-01-19T08:50:26.912+0000","rule":{"level":13,"description":"sshd: Attempt to login using a non-existent user","id":"5710","mitre":{"id":["T1110"],"tactic":["Credential Access"],"technique":["Brute Force"]},"firedtimes":3,"mail":true,"groups":["local","syslog","sshd","invalid_login","authentication_failed"],"pci_dss":["10.2.4","10.2.5","10.6.1"],"gpg13":["7.1"],"gdpr":["IV_35.7.d","IV_32.2"],"hipaa":["164.312.b"],"nist_800_53":["AU.14","AC.7","AU.6"],"tsc":["CC6.1","CC6.8","CC7.2","CC7.3"]},"agent":{"id":"000","name":"wazuh-manager"},"manager":{"name":"wazuh-manager"},"id":"1642582226.39554","full_log":"Nov  8 11:16:26 centos-manager1 sshd[20046]: invalid user from 172.17.1.1 port 39588","predecoder":{"program_name":"sshd","timestamp":"Nov  8 11:16:26","hostname":"centos-manager1"},"decoder":{"parent":"sshd","name":"sshd"},"data":{"srcport":"39588"},"location":"/var/log/auth.log"},"program":"active-response/bin/firewall-drop"}}

2022/01/19 08:50:26 active-response/bin/firewall-drop: Cannot read 'srcip' from data
```

But with a real alert (ssh):

```
** Alert 1642582342.43083: mail  - local,syslog,sshd,invalid_login,authentication_failed,pci_dss_10.2.4,pci_dss_10.2.5,pci_dss_10.6.1,gpg13_7.1,gdpr_IV_35.7.d,gdpr_IV_32.2,hipaa_164.312.b,nist_800_53_AU.14,nist_800_53_AC.7,nist_800_53_AU.6,tsc_CC6.1,tsc_CC6.8,tsc_CC7.2,tsc_CC7.3,
2022 Jan 19 08:52:22 (cible) 192.168.1.20->/var/log/auth.log

Rule: 5710 (level 13) -> 'sshd: Attempt to login using a non-existent user'
Src IP: 192.168.x.x
Src Port: 51852
Jan 19 08:52:22 cible sshd[1079]: Invalid user user from 192.168.x.xport 51852
```

No active response.
That's why I think the error comes from the log file, it is not filled in the right one, right?

**Comment by palaciosjeremias** [comment]
Hello Linow974,
Sorry for the late response!
I was searching for a similar issue and noticed that you are already working along Victor Rebollo in this google thread https://groups.google.com/g/wazuh/c/Q-H4oiQRD-c/m/UeoeNdZzAgAJ.
I'm getting in contact with him to work on a solution together.
I will get back ASAP.
Best regards.

**Comment by Linow974** [comment]
Hello Palacios, thank you for your message!

Indeed I have already made quite a bit of progress on my problem with
Victor, but I am still stuck with Active Response.

I posted my latest issue on Google Groups.

AR doesn't seem to work with certain alert syntax, yet it should work
because I've taken a manipulation from Victor with AR that works fine.

I hope to solve this problem!

Thanks for your help


Best regards



PS: sorry if my English is not good, I am of French origin

**Comment by palaciosjeremias** [comment]
Hello Linow974.
From the thread [here](https://groups.google.com/g/wazuh/c/Q-H4oiQRD-c/m/UeoeNdZzAgAJ) I understand that there were errors on the ruleset that wasn´t unchaining the expected result.
And, after a fresh install of Wazuh, it is finally working.
Let me know if this is the case. And if you have any further doubt please don´t hesitate to ask.

**Comment by Linow974** [comment]
Hello Palaces,

Thank you for your message.

Everything was back to normal after reinstalling.

I am now setting up my rules :)

Thanks again for the time you gave me!

**Comment by palaciosjeremias** [comment]
Glad to hear it´s working now!
We are going to close this issue then.
If you have any further doubt, please don´t hesitate to create a new issue or re-open this one if the topic is the same.
Best regards!
